no_proxy.go

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+type noProxy struct {
```

Comment:
Now that this file has been introduced, I think it might be a good idea to add a description in this file explaining what `no_proxy.go` is for.

The name does admittedly give a clue, but `noop_proxy.go` and `no_proxy.go` are pretty similar, as are the file contents apart from the crucial `start()` call below.

It might even be worth expanding on the brief comments in `noop_proxy.go` to add a descriptive header explaining that this is a dummy proxy whereas `no_proxy.go` provides the (mostly) empty proxy implementation required by vc, but which is specifically for systems that don't actually require a proxy process (namely kata containers environments running on systems whose host kernel supports vsock).

Reply:
@jodh-intel I fully agree! I'll add some more descriptions and more unit tests.

Reply:
Done